gems/aws-partitions/lib/aws-partitions/endpoint_provider.rb

```diff
@@ -38,9 +38,22 @@ def initialize(rules)
       # @param [String] sts_regional_endpoints [STS only] Whether to use `legacy` (global endpoint for
       #   legacy regions) or `regional` mode for using regional endpoint for supported regions
       #   except 'aws-global'
+      # @param [String] s3_us_east_1_regional_endpoint [S3 only] Whether to use
+      #   `legacy` (global endpoint) or `regional` mode for using regional
+      #   endpoint for S3 'us-east-1' region
       # @api private Use the static class methods instead.
-      def resolve(region, service, sts_regional_endpoints = nil)
-        "https://" + endpoint_for(region, service, sts_regional_endpoints)
+      def resolve(
+        region,
+        service,
+        sts_regional_endpoints = 'legacy',
+        s3_us_east_1_regional_endpoint = 'legacy'
```

Comment:
I see that you set defaults for both this, a different resolve method, and endpoint_for. Could we centralize them somewhere? If the default are ever changed, it must be done in multiple places.